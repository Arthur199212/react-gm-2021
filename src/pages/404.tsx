export { NotFoundPage as default } from '@app/features'
