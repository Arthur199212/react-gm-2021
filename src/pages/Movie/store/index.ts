export * from './movie.selectors'
export * from './movie.slice'
export * from './movie.thunks'
