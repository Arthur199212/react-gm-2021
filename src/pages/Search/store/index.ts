export * from './search.selectors'
export * from './search.slice'
export * from './search.thunks'
