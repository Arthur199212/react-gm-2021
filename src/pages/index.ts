export * from './Movie'
export * from './NotFound'
export * from './Search'
