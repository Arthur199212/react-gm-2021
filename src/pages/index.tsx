import { NextPageContext } from 'next'
import { useRouter } from 'next/router'

const IndexPage = () => {
  const router = useRouter()

  // Make sure we are in the browser
  if (typeof window !== 'undefined') {
    router.push('/search')
  }

  return <></>
}

export const getStaticProps = (ctx: NextPageContext) => {
  // We check for ctx.res to make sure we are on the server
  if (ctx.res) {
    ctx.res.writeHead(302, { Location: '/search' })
    ctx.res.end()
  }
  return { props: {} }
}

export default IndexPage
