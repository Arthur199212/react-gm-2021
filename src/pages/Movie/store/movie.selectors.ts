import { createSelector } from '@reduxjs/toolkit'
import { RootState } from '@app/store'
import { MovieState } from './movie.slice'

const selectMovieState = (state: RootState): MovieState => state.movie

export const selectMovie = createSelector(selectMovieState, state => state.movie)

export const selectMovieStatus = createSelector(selectMovieState, state => state.status)
