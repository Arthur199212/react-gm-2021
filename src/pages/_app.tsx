import type { AppProps } from 'next/app'
import Head from 'next/head'
import { Provider } from 'react-redux'
import { ErrorBoundary } from '@app/components'
import { createStore } from '@app/store'
import 'normalize.css'
import '@app/styles/index.scss'

const App = ({ Component, pageProps }: AppProps) => {
  const store = createStore(pageProps.preloadedState)

  return (
    <>
      <Head>
        <title>Movie Search App</title>
      </Head>
      <Provider store={store}>
        <ErrorBoundary>
          <Component {...pageProps} />
        </ErrorBoundary>
      </Provider>
    </>
  )
}

export default App
