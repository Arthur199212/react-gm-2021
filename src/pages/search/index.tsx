import { NextPageContext } from 'next'
import { fetchMoviesDefaultThunk } from '@app/features/SearchPage/store'
import { createStore } from '@app/store'

export { SearchPage as default } from '@app/features'

export const getServerSideProps = async (props: NextPageContext) => {
  const store = createStore()
  await store.dispatch(fetchMoviesDefaultThunk())

  return { props: { preloadedState: store.getState() } }
}
