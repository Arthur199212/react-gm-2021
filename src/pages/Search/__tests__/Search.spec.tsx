import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { Search } from '@app/pages'
import { store } from '@app/store'

describe('Search Page', () => {
  const setup = () =>
    render(
      <Provider store={store}>
        <Router>
          <Search />
        </Router>
      </Provider>
    )

  it('should render properly', () => {
    const { asFragment } = setup()

    expect(asFragment()).toMatchSnapshot()
  })
})
