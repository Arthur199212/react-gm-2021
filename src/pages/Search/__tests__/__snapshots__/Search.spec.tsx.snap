// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page should render properly 1`] = `
<DocumentFragment>
  <div
    class="app-search-page content"
  >
    <header
      class="header"
    >
      <div
        class="header-container container"
      >
        <a
          href="/"
        >
          <span
            class="title"
          >
            MovieSearchApp
          </span>
        </a>
        <a
          class="app-button"
          data-testid="header-button"
        >
          ADD MOVIE
        </a>
        <div
          class="app-modal-container"
          data-testid="modal-container"
        >
          <div
            class="app-modal"
          >
            <div
              class="app-dual-ring-spinner"
            />
          </div>
        </div>
      </div>
    </header>
    <main
      class="main"
    >
      <div
        class="search-box-container container"
      >
        <div
          class="search-header"
        >
          <h1
            class="title"
          >
            FIND YOUR MOVIE
          </h1>
        </div>
        <div
          class="search-container"
        >
          <input
            autocomplete="off"
            class="app-input-box"
            placeholder="What do you want to watch?"
            type="text"
            value=""
          />
          <a
            class="app-button"
          >
            SEARCH
          </a>
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="filters-container"
          data-testid="filters-container"
        >
          <ul
            class="app-tabs"
          >
            <li
              class="app-tab active"
            >
              all
            </li>
            <li
              class="app-tab"
            >
              comedy
            </li>
            <li
              class="app-tab"
            >
              crime
            </li>
            <li
              class="app-tab"
            >
              documentary
            </li>
            <li
              class="app-tab"
            >
              horror
            </li>
          </ul>
          <div
            class="dropdown-container"
          >
            <span
              class="label"
            >
              SORT BY
            </span>
            <div
              class="dropdown-menu"
            >
              <a
                class="app-button"
                data-testid="filters-dropdown-button"
              >
                release date
              </a>
              <ul
                class="app-dropdown"
                data-testid="filters-dropdown"
              >
                <li
                  class="app-dropdown-item"
                  data-testid="dropdown-item"
                >
                  rating
                </li>
                <li
                  class="app-dropdown-item"
                  data-testid="dropdown-item"
                >
                  release date
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="no-results-container"
        >
          Sorry, but nothing matched your search criteria.
        </div>
      </div>
    </main>
  </div>
  <footer
    class="app-footer"
  >
    MovieSearchApp
  </footer>
</DocumentFragment>
`;
